core/templates/components/forms/custom-forms-directives/edit-thumbnail-modal.component.ts

```diff
@@ -0,0 +1,153 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for edit thumbnail modal.
+ */
+
+import { trigger, transition, style, animate } from '@angular/animations';
+import { SimpleChanges } from '@angular/core';
+import { Component, Input, OnChanges } from '@angular/core';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { SvgSanitizerService } from 'services/svg-sanitizer.service';
+
+@Component({
+  selector: 'edit-thumbnail-modal',
+  templateUrl: './edit-thumbnail-modal.component.html',
+  animations: [
+    trigger('fade', [
+      transition('void => *', [
+        style({opacity: 0}),
+        animate(500, style({opacity: 1}))
+      ])
+    ])
+  ]
+})
+export class EditThumbnailModalComponent implements OnChanges {
+  @Input() bgColor: string;
+  @Input() uploadedImage: string | null;
+  @Input() aspectRatio: string;
+  @Input() previewDescription: string;
+  @Input() previewDescriptionBgColor: string;
+  @Input() previewFooter: string;
+  @Input() previewTitle: string;
+  @Input() allowedBgColors: string[];
+  @Input() tempBgColor: string;
+  @Input() dimensions: { height: number; width: number; };
+  @Input() openInUploadMode: boolean;
+  @Input() uploadedImageMimeType: string;
+
+  invalidImageWarningIsShown = false;
+  invalidTagsAndAttributes: {
+    tags: string[];
+    attrs: string[];
```

Comment:
are you trying to create a interface here?

Reply:
fixed